e2e tests broken in main branch.

Error as a reference:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x2472fdf]

goroutine 317 [running]:
./pkg/controllers/externalsecret.assertStoreIsUsable({0x3450738, 0xc00105c160})

```

It seems that this is a race condition where a store might still have no available status while a `ExternalSecret` is being reconciled. 

Issue can be reproduced by adding a delay in the `Validate()` method.

--- a/pkg/controllers/externalsecret/externalsecret_controller.go
+++ b/pkg/controllers/externalsecret/externalsecret_controller.go
@@ -483,7 +483,8 @@ func isSecretValid(existingSecret v1.Secret) bool {
 
 // assertStoreIsUsable assert that the store is ready to use.
 func assertStoreIsUsable(store esv1beta1.GenericStore) error {
-	if secretstore.GetSecretStoreCondition(store.GetStatus(), esv1beta1.SecretStoreReady).Status != v1.ConditionTrue {
+	condition := secretstore.GetSecretStoreCondition(store.GetStatus(), esv1beta1.SecretStoreReady)
+	if condition == nil || condition.Status != v1.ConditionTrue {
 		return fmt.Errorf(errSecretStoreNotReady, store.GetName())
 	}
 	return nil

